tests/Aspire.Hosting.Testing.Tests/Aspire.Hosting.Testing.Tests.csproj

```diff
@@ -15,13 +15,22 @@
 
   <ItemGroup>
     <Compile Include="$(SharedDir)KnownConfigNames.cs" Link="Utils\KnownConfigNames.cs" />
+
+    <Compile Include="$(RepoRoot)tests\Aspire.Hosting.Tests\Utils\LoggerNotificationExtensions.cs" />
+    <Compile Include="$(RepoRoot)tests\Aspire.Hosting.Tests\Utils\MSBuildUtils.cs" LinkBase="shared\tests\MSBuildUtils.cs" />
+    <Compile Include="$(RepoRoot)tests\testproject\Common\TestResourceNames.cs" LinkBase="shared\tests\TestResourceNames.cs" />
+    <Compile Include="$(TestsSharedDir)Logging\*.cs" LinkBase="shared\Logging" />
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\Aspire.Components.Common.Tests\Aspire.Components.Common.Tests.csproj" />
     <ProjectReference Include="..\..\src\Aspire.Hosting.AppHost\Aspire.Hosting.AppHost.csproj" IsAspireProjectResource="false" />
     <ProjectReference Include="..\..\src\Aspire.Hosting.Testing\Aspire.Hosting.Testing.csproj" IsAspireProjectResource="false" />
-    <ProjectReference Include="..\Aspire.Hosting.Tests\Aspire.Hosting.Tests.csproj" />
     <ProjectReference Include="..\TestingAppHost1\TestingAppHost1.AppHost\TestingAppHost1.AppHost.csproj" />
+
+    <!-- This ProjectReference is *only* to get this project built, but not referenced directly.
+         It will be loaded dynamically by individual tests. -->
+    <ProjectReference Include="..\testproject\TestProject.AppHost\TestProject.AppHost.csproj" ReferenceOutputAssembly="false" IsAspireProjectResource="false" Private="false" />
```

Comment:
I don't get why this is needed. How does it allow the project to build?

Reply:
This adds a project dependency, so msbuild will ensure that the project is built with the test project.
- The `ReferenceOutputAssembly=false` causes the output assembly from `TestProject` to not participate in the `Compile` for `Aspire.Hosting.Testing.Tests`.
- `IsAspireProjectResource=false` - we don't want to have code generated for `TestProject`, so that the test can load the assembly from disk instead.
- `Private=false` - ensures that none of the files (the `.deps.json` and some other files) are copied to `Aspire.Hosting.Testing.Tests` `bindir`.

Reply:
continued: And this allows `Aspire.Hosting.Testing.Tests` assembly to have no "ties" to `TestProject`, and the test can cleanly load the assembly via reflection.